src/OpenColorIO/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+set(SOURCES
+	Baker.cpp
+	BitDepthUtils.cpp
+	Caching.cpp
+	ColorSpace.cpp
+	Config.cpp
+	Context.cpp
+	Display.cpp
+	Exception.cpp
+	fileformats/FileFormat3DL.cpp
+	fileformats/FileFormatCCC.cpp
+	fileformats/FileFormatCC.cpp
+	fileformats/FileFormatCDL.cpp
+	fileformats/FileFormatCSP.cpp
+	fileformats/FileFormatDiscreet1DL.cpp
+	fileformats/FileFormatHDL.cpp
+	fileformats/FileFormatICC.cpp
+	fileformats/FileFormatIridasCube.cpp
+	fileformats/FileFormatIridasItx.cpp
+	fileformats/FileFormatIridasLook.cpp
+	fileformats/FileFormatPandora.cpp
+	fileformats/FileFormatResolveCube.cpp
+	fileformats/FileFormatSpi1D.cpp
+	fileformats/FileFormatSpi3D.cpp
+	fileformats/FileFormatSpiMtx.cpp
+	fileformats/FileFormatTruelight.cpp
+	fileformats/FileFormatVF.cpp
+	GpuShader.cpp
+	GpuShaderDesc.cpp
+	GpuShaderUtils.cpp
+	HashUtils.cpp
+	ImageDesc.cpp
+	ImagePacking.cpp
+	Logging.cpp
+	Look.cpp
+	LookParse.cpp
+	MathUtils.cpp
+	md5/md5.cpp
+	OCIOYaml.cpp
+	Op.cpp
+	OpOptimizers.cpp
+	ops/Allocation/AllocationOp.cpp
+	ops/Exponent/ExponentOps.cpp
+	ops/Log/LogOps.cpp
+	ops/Lut1D/Lut1DOp.cpp
+	ops/Lut3D/Lut3DOp.cpp
+	ops/Matrix/MatrixOps.cpp
+	ops/NoOp/NoOps.cpp
+	ops/Truelight/TruelightOp.cpp
+	OpTools.cpp
+	ParseUtils.cpp
+	PathUtils.cpp
+	Platform.cpp
+	Processor.cpp
+	ScanlineHelper.cpp
+	Transform.cpp
+	transforms/AllocationTransform.cpp
+	transforms/CDLTransform.cpp
+	transforms/ColorSpaceTransform.cpp
+	transforms/DisplayTransform.cpp
+	transforms/ExponentTransform.cpp
+	transforms/FileTransform.cpp
+	transforms/GroupTransform.cpp
+	transforms/LogTransform.cpp
+	transforms/LookTransform.cpp
+	transforms/MatrixTransform.cpp
+	transforms/TruelightTransform.cpp
+	UnitTest.cpp
+)
+
+add_library(OpenColorIO ${SOURCES})
+set_property(TARGET OpenColorIO PROPERTY CXX_STANDARD 11)
+target_include_directories(OpenColorIO
+	PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+target_link_libraries(OpenColorIO 
+	PUBLIC 
+		public_api
+	PRIVATE
+		tinyxml
+		yamlcpp
+		pystring
+		sampleicc
+)
+if(NOT BUILD_SHARED_LIBS)
+	target_compile_definitions(OpenColorIO
+		PRIVATE
+		OpenColorIO_SKIP_IMPORTS
+	)
+endif()
+
+if(OCIO_USE_SSE)
+	target_compile_definitions(OpenColorIO
+		PRIVATE
+			USE_SSE
+	)
+endif()
+if(MSVC)
+	# Because our Exceptions derive from std::runtime_error we can safely disable this warning
```

Comment:
That's an unfortunate side-effect of deriving the OCIO exception from STL exceptions. The STL library classes were never designed to be used as base classes and exported. At the same time, it's extremely an useful design to ease the code writing.

Reply:
That's an extremely common use case, some would say it's what std::runtime_error was made for. It's kinda silly that the MSVC STL has such flimsy binary compatability that this would warrant a warning.
We can only hope that this is mitigated by developers only building this library in the same configuration as the app it will end up embedded in.

Reply:
However, it's a quite risky assumption as VS could do lot of things in your back. When there are some memory model mismatches between libraries, it could change it without asking i.e. only a command line warning.